Packs/GitLab/Integrations/GitLabEventCollector/GitLabEventCollector.py

```diff
@@ -1,179 +1,285 @@
-    for event in events:
-        if 'details' in event:
-            for action in ['add', 'change', 'remove']:
-                if action in event['details']:
-                    event['details']['action'] = f'{action}_{event["details"][action]}'
-                    event['details']['action_type'] = action
-                    event["details"]['action_category'] = event['details'][action]
-                    break
-    return events
-
-
-def main() -> None:  # pragma: no cover
-    demisto_params = demisto.params() | demisto.args()
-    url = urljoin(demisto_params['url'], '/api/v4/')
-    should_push_events = argToBoolean(demisto_params.get('should_push_events', 'false'))
-    events_collection_management = {
-        'groups_ids': argToList(demisto_params.get('group_ids', '')),
-        'projects_ids': argToList(demisto_params.get('project_ids', '')),
-        'event_types': ['groups', 'projects']
-    }
-    headers = {'PRIVATE-TOKEN': demisto_params.get('api_key', {}).get('password')}
-    verify = not demisto_params.get('insecure', 'True')
-    request_object = {
-        'method': Method.GET,
-        'url': url,
-        'headers': headers,
-        'verify': verify
-    }
-    last_run = demisto.getLastRun()
-    if 'groups' not in last_run and 'projects' not in last_run and 'events' not in last_run:
-        last_run = dateparser.parse(demisto_params['after'].strip()).isoformat()  # type: ignore
-        last_run = {
-            'groups': last_run,
-            'projects': last_run,
-            'events': last_run,
-        }
 
-    options = IntegrationOptions(**demisto_params)
+def fetch_events_command(client: Client, params: dict, last_run: dict, events_types_ids: dict) -> tuple[list, list, dict]:
+    """
+    Collects log events from GitLab using pagination.
 
-    request = IntegrationHTTPRequest(**request_object)
+    Args:
+        client (Client): the client implementing the API to GitLab.
+        params (dict): the instance configuration parameters.
+        last_run (dict): the lastRun object, holding information from the previous run.
+        events_types_ids (dict): The groups / projects Ids to fetch events for.
 
-    client = Client(request, options, last_run)
+    Returns:
+        (list) the audit events retrieved from the API call.
+        (list) the groups and projects events retrieved from the API call.
+        (dict) the updated lastRun object.
+    """
 
-    get_events = GetEvents(client, options)
+    query_params_url_suffix = prepare_query_params(params, last_run["audit_events"])
+    audit_events = client.fetch_events(query_params_url_suffix, last_run["audit_events"], params)
+    demisto.debug(f"Aggregated audits events: {len(audit_events)}")
+
+    group_and_project_events = []
+    for event_type in ['groups', 'projects']:
+        for obj_id in events_types_ids.get(f'{event_type}_ids', []):
+            client._base_url = f"{params['url']}/api/v4/{event_type}/{obj_id}/audit_events"
```

Comment:
must we change _base_url? best practice is to leave it as-is, only the suffix should be sent